charts/tfy-karpenter-config/values.yaml

```diff
@@ -157,7 +157,108 @@ karpenter:
     extraTags: {}
     ## @skip karpenter.defaultNodeTemplate.userData
     # Set this to "" to disable this injection
-    userData: ""
+    userData: |
+      #!/bin/bash
+      set -ex
+
+      version_lte() {
+          printf '%s\n' "$1" "$2" | sort -C -V
+      }
+
+      version_lt() {
+          ! version_lte "$2" "$1"
+      }
+
+      CONTAINERD_VERSION_OUTPUT="$(containerd --version)"
+      IFS=" " read -a  CONTAINERD_VERSION_STRING <<< "${CONTAINERD_VERSION_OUTPUT}"
+      CONTAINERD_VERSION=${CONTAINERD_VERSION_STRING[2]}
+      echo containerd version "${CONTAINERD_VERSION}"
+
+      setup_soci() {( set -ex
+        yum install fuse -y
+        modprobe fuse
+
+        curl \
+        --silent \
+        --show-error \
+        --retry 10 \
+        --retry-delay 1 \
+        -L https://github.com/awslabs/soci-snapshotter/releases/download/v0.4.0/soci-snapshotter-0.4.0-linux-amd64.tar.gz \
+        -o soci-snapshotter-0.4.0-linux-amd64.tar.gz
+
+        tar -C /usr/local/bin -xvf soci-snapshotter-0.4.0-linux-amd64.tar.gz soci soci-snapshotter-grpc
+        rm soci-snapshotter-0.4.0-linux-amd64.tar.gz
+
+        cat > /etc/systemd/system/soci-snapshotter.service << EOF
+      [Unit]
+      Description=soci snapshotter containerd plugin
+      Documentation=https://github.com/awslabs/soci-snapshotter
+      After=network.target
+      Before=containerd.service
+
+      [Service]
+      Type=notify
+      ExecStart=/usr/local/bin/soci-snapshotter-grpc
+      Restart=always
+      RestartSec=5
+
+      [Install]
+      WantedBy=multi-user.target
+      EOF
+
+        systemctl daemon-reload
+        systemctl enable --now soci-snapshotter
+        systemctl status soci-snapshotter
+
+        echo patching /etc/eks/containerd/containerd-config.toml
+        cat > /etc/eks/containerd/containerd-config.toml << EOF
+      version = 2
+      root = "/var/lib/containerd"
+      state = "/run/containerd"
+
+      [grpc]
+      address = "/run/containerd/containerd.sock"
+
+      [plugins."io.containerd.grpc.v1.cri".containerd]
+      default_runtime_name = "runc"
+      discard_unpacked_layers = true
+      ## ENABLE SOCI SNAPSHOTTER
+      snapshotter = "soci"
+      disable_snapshot_annotations = false
+      ##
+
+      [plugins."io.containerd.grpc.v1.cri"]
+      sandbox_image = "602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/pause:3.5"
```

Comment:
This should not be hardcoded here!, instead use the pause template variable here and let bootstrap script replace this according to the region

Reply:
Yeah, my bad. I found the logic on https://github.com/awslabs/amazon-eks-ami/blob/master/files/bootstrap.sh